### DISCUSSION TITLE
Implementing conditional behaviour to useAction hook

### DISCUSSION THREAD

**Discussion by merthanmerter** [next-safe-action/next-safe-action#266] [Ideas]
Some may think server actions are not meant to be fetching client side data but I personally find them useful when combined with middlewares and revalidating data with state hooks instead of search queries etc.

That said, would it be possible to introduce something like  "shouldFetch / _shouldTrigger_" parameter to conditionally trigger the action with useAction hook? 

```ts
const { execute, result, status } = useAction(action, { executeOnMount: { input, shouldTrigger: !!condition } })

// // React.useEffect(() => {
// // 	if (!condition) return
// // 	execute()
// // }, [condition])
```

By the way, this works;
```ts
const { data, error, isLoading } = useSWR(shouldFetch ? 'action' : null, () => action({}), {})
```

**Comment by TheEdoRan** [comment]
Hi, as you said, they're not really meant to fetch data, route handlers should be used for that instead. Though, since `executeOnMount` functionality is already implemented, I could add something like an `executeWhen` property to trigger execution.

**Reply by sp88011** [reply]
+1 to this idea. Would love to use this library for simple data fetching, and this seems like the way to do it.

**Reply by ScreamZ** [reply]
I'm voting to deprecate `executeOnMount`. As you said, server actions are not meant to fetch data and should be the result of actions or side effects.

The `executeOnMount` feature can easily be coded with a `useEffect` if you wish so.

If you wish to play on client side code like for an SPA, you should rather relies on SWR or tanstack query which is way more optimised for the use case you indicate.

**Reply by merthanmerter** [reply]
> I'm voting to deprecate `executeOnMount`. As you said, server actions are not meant to fetch data and should be the result of actions or side effects.
> 
> 
> 
> The `executeOnMount` feature can easily be coded with a `useEffect` if you wish so.

makes more sense tbh 😀

**Reply by TheEdoRan** [reply]
> I'm voting to deprecate `executeOnMount`.

Yeah, after thinking about it I agree with you @ScreamZ. Actions should not have an `executeOnMount` functionality. We'll see if this will change with React Server Functions. I'll keep the functionality for now, but deprecate it.

**Reply by merthanmerter** [reply]
I still think some kind of shouldTrigger functionality would be still useful though. Maybe not for data fetching but it would be a solid way to block the the action to be triggered instead of adding extra return layer.

**Reply by TheEdoRan** [reply]
@merthanmerter I think it's better to let the user handle that, in my opinion it's not a library job.
